pkg/apis/triggers/v1alpha1/trigger_validation.go

```diff
@@ -70,21 +70,25 @@ func (t TriggerSpecTemplate) validate(ctx context.Context) *apis.FieldError {
 }
 
 func (t triggerSpecBindingArray) validate(ctx context.Context) *apis.FieldError {
-	if len(t) > 0 {
-		for i, b := range t {
-			// Either Ref or Spec should be present
-			if b.Ref == "" && b.Spec == nil {
-				return apis.ErrMissingOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Spec", i))
-			}
-
-			// Both Ref and Spec can't be present at the same time
-			if b.Ref != "" && b.Spec != nil {
+	for i, b := range t {
+		switch {
+		case b.Ref != "":
+			switch {
+			case b.Spec != nil: // Cannot specify both Ref and Spec
 				return apis.ErrMultipleOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Spec", i))
-			}
-
-			if b.Ref != "" && b.Kind != NamespacedTriggerBindingKind && b.Kind != ClusterTriggerBindingKind {
+			case b.Name != "": // Cannot specify both Ref and Name
+				return apis.ErrMultipleOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Name", i))
```

Comment:
I can able to specify binding like below

```
apiVersion: triggers.tekton.dev/v1alpha1
kind: EventListener
metadata:
  name: github-listener-interceptor
spec:
  triggers:
    - name: github-listener
      interceptors:
        - github:
            secretRef:
              secretName: github-secret
              secretKey: secretToken
            eventTypes:
              - pull_request
        - cel:
            filter: "body.action in ['opened', 'synchronize', 'reopened']"
      bindings:
        - ref: github-pr-binding
        - name: gitrevision1
          value: $(body.pull_request.head.sha)
        - name: gitrepositoryurl1
          value: $(body.repository.clone_url)
```
and EL creation is success :thinking: 

But as per the validation it should not 
correct me if my understanding is wrong